src/data_source/storage/sql.rs

```diff
@@ -774,6 +772,62 @@ where
         let count: i64 = row.get(0);
         Ok(count as usize)
     }
+
+    async fn sync_status(&self) -> QueryResult<SyncStatus> {
```

Comment:
Are we worried at all about the possibility of spamming our nodes with this endpoint? Looks expensive

Reply:
Yes. But it shouldn't be *that* expensive. It's not the only endpoint we have that currently triggers a full table scan. I think we can optimize this to make it less expensive (will do this after we have seen the behavior in a realistic setting with large tables). In the long run, we probably want to implement some type of API credit system, where endpoints like this would require more credit than endpoints that are easier to index